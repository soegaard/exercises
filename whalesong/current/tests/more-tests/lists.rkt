#lang planet dyoo/whalesong/base

'(1 2 3)
(list "hello" "world")

(list? empty)
(list? '(1))
(list? '(1 2))
(list? 1)

(empty? empty)
(empty? '())
(cons? '())
(cons? '(hello))
(first '(hello))
(rest '(hello))
(second '(hello world))

(andmap even? '(2 4 6 8))
(andmap even? '())
(andmap even? '(2 4 5 8))
(andmap even? '(5))
(andmap even? '(1 3 5 7))
(andmap even? '(1 3 8 7))

(ormap even? '(2 4 6 8))
(ormap even? '())
(ormap even? '(2 4 5 8))
(ormap even? '(5))
(ormap even? '(1 3 5 7))
(ormap even? '(1 3 8 7))


(vector->list #(3 1 4))
