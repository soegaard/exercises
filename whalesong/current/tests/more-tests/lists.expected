(1 2 3)
("hello" "world")
true
true
true
false
true
true
false
true
hello
()
world
true
true
false
false
false
false
true
false
true
false
false
true
(3 1 4)
